capstone-frontend/src/components/ListPage.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+
+class ListPage extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      gBooks: [],
+      loading: true,
+      empty: false
+    }
+  }
+
+
+  fetchBooks = async () => {
+
+    const userID = window.localStorage.getItem("userID");
+
+    console.log("Fetch Books LIST PAGE")
+
+    const bookLists = await fetch(`/api/booklist?userID=${userID}`, {
+      method: "GET",
+    }).then(resp => resp.json());
+
+    console.log("Booklist", bookLists)
+
+    const bookList = Array.from(bookLists).filter(bookList => (bookList.id === this.props.match.params.id))[0].gbookIDs;
+
+    console.log("Booklist", bookList)
+
+    if (bookList.length === 0) {
+      this.setState({loading: false, empty: true});
+      return;
+    }
+
+    const gBooks = [];
+
+    await Promise.all(bookList.map(async (gBookID) => {
```

Comment:
Is this what causes the latency in rendering the list? Instead of awaiting for all the books to be grabbed from the Google Books API to render the list, could you leave this as truly asynchronous (ie no `await`) and render the list as they come? Not sure if that would fix your problem and eliminate the need for the Loading Page.

Reply:
Also worried about what would happen in the case of longer lists. Say I have 100 books on my list, it would probably be better to render the books as they come than all at once.